ds701_book/20-Recommender-Systems.qmd

```diff
@@ -362,39 +488,75 @@ These properties interfere with similarity assessment.
 Bias correction is crucial for CF recommender systems.
 
 We need to include
+
 * Per-user offset
 * Per-item offset
 * Global offset
 
+## Representing biases
+
 Hence we need to form a per-item bias of:
     
-$$ b_{ui} = \mu + \alpha_u + \beta_i $$
+$$ 
+b_{ui} = \mu + \alpha_u + \beta_i 
+$$
+
+where 
+
+* $\alpha_u$ is the per-user offset of user $u$ and 
+* $\beta_i$ is the per-item offset of item $i$.
 
-where $\alpha_u$ is the per-user offset of user $u$ and $\beta_i$ is the per-item offset of item $i$.
 
-How can we estimate the $\alpha$s, the $\beta$s, and the $\mu$?
+## Estimating biases
+
+::: {.content-visible when-profile="slides"}
+$$ 
+b_{ui} = \mu + \alpha_u + \beta_i 
+$$
+:::
+
+How can we estimate the $\alpha\textnormal{s}$, the $\beta\textnormal{s}$, and
+the $\mu\textnormal{s}$?
 
 Let's assume for a minute that we had a fully-dense matrix of ratings $R$.
 
 $R$ has items on the rows and users on the columns.
 
 Then what we want to estimate is
 
-$$\min_{\alpha,\beta,\mu} \Vert R - \mathbf{1}\alpha^T + \beta\mathbf{1}^T + \mu1\Vert^2 + \lambda(\Vert\alpha\Vert^2 + \Vert\beta\Vert^2) $$
+$$
+\min_{\alpha,\beta,\mu} 
+\Vert R - \mathbf{1}\alpha^T + \beta\mathbf{1}^T + \mu1\Vert^2 
++ \lambda(\Vert\alpha\Vert^2 + \Vert\beta\Vert^2) 
+$$
 
 Here, $\mathbf{1}$ represents appropriately sized vectors of ones, and $1$ is a matrix of ones.
```

Comment:
Can we define dimensions for this problem?

Reply:
added dimensions where in previous slide where alpha and beta is introduced.